status/views.py

```diff
@@ -0,0 +1,107 @@
+"""
+Status checks for LORE.
+
+Notes:
+
+* Useful messages are logged, but "no config found" is returned whether
+  settings are missing or invalid, to prevent information leakage.
+* Different services provide different information, but all should return
+  "up," "down", or "no config found" for the "status" key.
+"""
+
+import re
+from datetime import datetime
+import logging
+
+from django.conf import settings
+from django.http import JsonResponse
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    try:
+        conf = settings.DATABASES['default']
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": "no config found"}
+    try:
+        start = datetime.now()
```

Comment:
After adding a context manager, it results in more complexity than it's worth. The context manager returns a value on `__enter__`, not `__exit__`, but it must have an `__exit__` method or it's invalid. The timer would be kicked off in the `__enter_`, and that's where the value must be returned from, so it gets ugly and it seems like this isn't actually a good use for a context manager.

If you think I'm missing the approach you had in mind, please send a link to a context manager that works the way you were thinking and I'll give it another go.